2do_fe/src/components/PopUps/AddTask.js

```diff
@@ -1,24 +1,165 @@
+  
+  const [taskData, setTaskData] = useState({
+    status: "in progress",
+    title: "",
+    description: "",
+    dueDate: "",
+  });
+
+  const [dateValue, setDate] = useState(null);
+
+  const dispatch = useDispatch();
+
+  const handleDateChange = (newDateValue) => {
+    setDate(newDateValue);
+    setTaskData({
+      ...taskData, 
+      dueDate: newDateValue,});
+  };
+
+  const handleChange = (e) => {
+    setTaskData({ 
+      ...taskData, [e.target.name]: e.target.value,
+     });
+  };
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    props.setTrigger_1(false);
+    dispatch(taskActions.addTask(taskData));
+  }
+
+
+  return (
+    <center>
+      <div id="Form-container">
+        <form onSubmit={handleSubmit}>
+          {/* TITLE INPUT*/}
+          <label htmlFor="title">Title</label>
+          <TextField
+            required
+            className="textfieldstyle"
+            type="text"
+            name="title"
+            value={taskData.title}
+            placeholder="Enter task title"
+            onChange={handleChange}
+          />
+
+          {/* DESCRIPTION INPUT */}
+          <label htmlFor="description">Description</label>
+          <TextField
+            className="textfieldstyle"
+            type="text"
+            name="description"
+            value={taskData.description}
+            placeholder="Enter task description"
+            multiline
+            rows={4}
+            onChange={handleChange}
+          />
+          
+          {/* DATE INPUT */}
+          <label htmlFor="date">Due Date</label>
+          <LocalizationProvider dateAdapter = {AdapterDayjs}>
+            <DesktopDatePicker
+              //label = "Due Date"
+              inputFormat = "MM/DD/YYYY"
+              value = {dateValue}
+              onChange = {handleDateChange}
+              renderInput = {
+                (params) => 
+                <TextField 
+                name = "dueDate"
+                {...params}
+                className="textfieldstyle" 
+                id="duedate"
+                />
+              }
+              />
+          </LocalizationProvider>
+          
+
+          <div className="Button-container">
+            <button
+              type = "submit"
+              className="Add-button"
+            >
+              Confirm
+            </button>
+            <button onClick={()=>props.setTrigger_1(false)} className="Cancel-button">
+              Cancel
+            </button>
+          </div>
+        </form>
+      </div>
+    </center>
+  );
+}
 
 function AddTask() {
+  const [isOpen, setIsOpen] = useState(false);
+
   return (
     <>
       <div className="grid grid-cols-5 gap-3">
         <div className="py-3 mb-2 justify-self-start pl-5 flex col-span-2 w-80 items-center ">
           <img
             src={AddTaskIcon}
             alt="Add task icon"
-            onClick={openTaskDesc}
+            // onClick={openTaskDesc}
```

Comment:
can we remove  // onClick={openTaskDesc}

Reply:
On it!